src/datasets/arrow_dataset.py

```diff
@@ -2822,16 +2874,30 @@ def map(
                 drop_last_batch=drop_last_batch,
                 remove_columns=remove_columns,
                 keep_in_memory=keep_in_memory,
-                load_from_cache_file=load_from_cache_file,
                 cache_file_name=cache_file_name,
                 writer_batch_size=writer_batch_size,
                 features=features,
                 disable_nullable=disable_nullable,
                 fn_kwargs=fn_kwargs,
                 new_fingerprint=new_fingerprint,
-                disable_tqdm=disable_tqdm,
-                desc=desc,
             )
+            try:
+                transformed_dataset = load_processed_shard_from_cache(dataset_kwargs)
+            except NonExistentDatasetError:
+                for rank, done, content in Dataset._map_single(**dataset_kwargs):
+                    if done:
+                        shards_done += 1
+                        if pbar is None:
+                            init_pbar()
+                        pbar.set_description(f"Processing the dataset ({shards_done}/{num_shards} shards)")
+                        logger.debug(f"Finished processing shard number {rank} of {num_shards}.")
+                        (transformed_dataset,) = content
```

Comment:
I think you can simplify this, no need to use a tuple
```suggestion
                        transformed_dataset = content
```